qiskit_machine_learning/algorithms/classifiers/vqc.py

```diff
@@ -53,18 +53,27 @@ def __init__(
         loss: str | Loss = "cross_entropy",
         optimizer: Optimizer | None = None,
         warm_start: bool = False,
-        quantum_instance: QuantumInstance | None = None,
+        quantum_instance: QuantumInstance | Backend | None = None,
```

Comment:
No, in this case we will break the interface. As an example:
```python
def fun1(a, b, c):
    print(a, b, c)

def fun2(a, c, b):
    print(a, b, c)

fun1(1, 2, 3)
fun2(1, 2, 3)
```
The parameters are not key-value, so the parameter order is important. What is `c` in both cases?
We can introduce:
```python
def fun1(a, *, b, c):
    print(a, b, c)
```
Then we can swap `b` and `c` in the definition as they are key-value parameters now and you may call only via `fun1(1, b=2, c=3)`

Reply:
FYI It was the last argument when VQC was first released. But since then more capability was added and the additional defaulted parameters (initial_point and callback) got included at the end to ensure compatibility with earlier versions. Had we had some sort level of value versus keyword parameter enforcement on the API, as @adekusar-drl points out, moving somethings around then becomes a possibility.

Reply:
Ok, let's keep it as is.